dlt/common/configuration/providers/toml.py

```diff
@@ -179,36 +182,37 @@ def _read_toml_file(self, toml_path: str) -> tomlkit.TOMLDocument:
             return None
 
     def _read_toml_files(
-        self, name: str, file_name: str, toml_path: str, global_path: str
+        self, name: str, file_name: str, toml_paths: List[str]
     ) -> tomlkit.TOMLDocument:
+        """Merge all toml files into one"""
+
         try:
-            if (project_toml := self._read_toml_file(toml_path)) is not None:
-                pass
-            elif (project_toml := self._read_google_colab_secrets(name, file_name)) is not None:
-                pass
-            elif (project_toml := self._read_streamlit_secrets(name, file_name)) is not None:
-                pass
-            else:
-                # empty doc
-                project_toml = tomlkit.document()
-            if global_path:
-                global_toml = self._read_toml_file(global_path)
-                if global_toml is not None:
-                    project_toml = update_dict_nested(global_toml, project_toml)
-            return project_toml
+            # merge all toml files into one
+            result_toml: Optional[tomlkit.TOMLDocument] = None
+            for path in toml_paths:
+                if (loaded_toml := self._read_toml_file(path)) is not None:
+                    if result_toml is None:
+                        result_toml = loaded_toml
+                    else:
+                        result_toml = update_dict_nested(loaded_toml, result_toml)
+
+            # if nothing was found, try to load from google colab or streamlit
```

Comment:
this part has changed in a way that colab and streamlit secrets only are loaded if no toml files at all are found. In the old implementation a potential global setting was merged into these. I am not sure if this is ok, this should definately be clear before we merge this.